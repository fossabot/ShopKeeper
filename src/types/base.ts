// Copyright (C) 2018 Sleep E-Z USA, Inc. / Evan Darwin
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import {Required} from '../lib/typing';
import {ProductImageItem} from './shopify/image';
import {Metafield} from './shopify/metafield';
import {Store} from './store';

/**
 * These types define special strings used for reading Shopify
 * responses, as well as various other uses throughout the
 * system. Each [[ShopifyType]] class must provide an `itemType`
 * containing one of these enum values.
 */
export enum ShopifyTypes {
  Page = 'pages',
  Option = 'product_option',
  Variant = 'variants',
  ProductImage = 'product_image',

  Product = 'products',
  Country = 'countries',
  Metafield = 'metafield',
  Province = 'province',
  Redirect = 'redirect',
  Shop = 'shop',
}

/**
 * The ShopifyType is one of the foundational types for ShopKeeper.
 * All entities that can be managed with the ShopKeeper framework
 * must extend this class to maintain simplicity and interoperability.
 *
 * @typeparam R   This generic describes the shape of the incoming data
 *                as returned from Shopify's Admin API.
 *
 * @typeparam U   This generic describes the shape of the incoming data
 *                as provided locally by the user (it may not exist remotely)
 */
export abstract class ShopifyItem<R extends RawShopifyData,
                                            U extends UserData> {
  abstract itemType: ShopifyTypes;
  includesRemote: boolean;
  store: Store;
  data: Indexable&Partial<{handle: Handle}>&Required<U>;
  protected rawData: R;
  protected userData: U|null;

  constructor(store: Store, data: R, remote = false) {
    this.store = store;
    this.includesRemote = remote;
    this.rawData = data;
  }

  id(): number {
    return this.rawData.id || this.data.id || -1;
  }

  isResolved(): boolean {
    return (this.id() > -1);
  }

  addUserData(data: U) {
    this.userData = data;
  }

  toShopifyJSON(): R {
    return this.mergeToRaw();
  }

  abstract mergeToRaw(): R;
}

export abstract class LocalShopifyItem<D extends UserData> {
  abstract itemType: ShopifyTypes;
  abstract data: D;

  constructor(data: D) {
    this.data = data;
  }
}

/**
 * This is data given by the user to re-create the user's
 * intended Shopify item. This means that this type will
 * likely have optional fields as the user should be allowed
 * to define properties as they please (except required fields).
 *
 * **NOTE**: If the `id` field is provided in the user's config,
 * it will simply be ignored and replaced by the remote value.
 * Concurrently, the user should *never* define the `id` value
 * as it will differ between environments.
 */
export type UserData = {};

/**
 * This type is used to represent the raw JSON response as received
 * from Shopify. Since it is returned from Shopify, this **`id`**
 * value will be the preferred reference method.
 */
export type RawShopifyData = Indexable&{};
export type ShopifyData = Indexable&UserData;

export interface ShopifyType<D, I> { new(store: Store, data: D): I; }

/**
 * We specify a type to allow uniqueness among
 * types carrying a `handle` property.
 */
export type Handle = string;

/**
 * All Shopify types available in the scope of
 * ShopKeeper are referenced based on their ID
 * available from Shopify.
 */
export type Indexable = {
  id: number,
};

/**
 * The following Shopify types carry the 'handle'
 * element: Article, Blog, CustomCollection,
 *          SmartCollection, Page, Product, Blog
 */
export type Handleable<H extends Handle> = {
  handle: H,
};

export type Purchasable = {
  price?: number,
  comparePrice?: number
};

export type Purchasable$Raw = {
  price: number|null,
  compare_at_price: number|null
};

/**
 * The Shopify type provides a `title` field for
 * the name of the element. If available, the field
 * is always required.
 */
export type Titled = {
  title: string
};

export type Metafields = {
  metafields: Array<Metafield<string>>
};
export type Metafields$Raw = Metafields;

/**
 * The type is positional and thus can be given a
 * specified position during submission. No actions
 * are required in the user's projects to position
 * these besides re-ordering their array elements.
 */
export type Positional = Partial<Positional$Raw>;
export type Positional$Raw = {
  position: number
};

/**
 * The element carries a timestamp of when the
 * element was created.
 */
export type Date$CreatedAt = {
  createdAt: Date
};
export type Date$CreatedAt$Raw = {
  created_at: string
};

/**
 * The element carries a timestamp of when the
 * element was last updated. This value can
 * sometimes be modified after the fact.
 */
export type Date$UpdatedAt = {
  updatedAt: Date
};
export type Date$UpdatedAt$Raw = {
  updated_at: string
};

/**
 * The element carries a timestamp of when the
 * element was last published. This value can be
 * edited after the fact.
 */
export type Date$PublishedAt = {
  publishedAt?: Date|null
};
export type Date$PublishedAt$Raw = {
  published_at: string|null
};

/**
 * The element accepts simple HTML content in
 * the form of a string.
 */
export type Content$HTML = {
  html?: string|'',
  template?: string|null,
};

export type Content$HTML$Raw = {
  body_html: string
};

/**
 * The element accepts binary or other formatted data
 * and can be submitted to Shopify's servers
 */
export type Content$Binary = {
  path?: string,
  filename?: string
};
export type Content$Binary$Raw = {
  attachment?: string,
  filename?: string
};

/**
 * If the element allows for HTML we can specify a
 * template to render and replace the `body_html`
 * content. If the `template` parameter is specified
 * then the `body_html` will be overridden.
 */
export type Content$Template = {
  template?: string
};
export type Content$Template$Raw = {
  body_html: string
};

export type Image$Single = {
  image?: ProductImageItem|null
};
export type Image$Single$Raw = {
  image_id?: number
};

export type Image$Multiple = {
  images?: ProductImageItem[]
};
export type Image$Multiple$Raw = {
  images?: number[]
};