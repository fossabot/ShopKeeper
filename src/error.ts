// Copyright (C) 2018 Sleep E-Z USA, Inc. / Evan Darwin
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import {ShopifyLogger} from './logging';
import {colorPath, colorType} from './theme';
import {ShopifyTypes} from './types/base';

const logging = new ShopifyLogger();

export class ConfigError extends Error {}

export function errNoValidConfig() {
  return new ConfigError(
      `This command requires a valid configuration. ` +
      `Verify '${colorPath('shopify.config.js')}' exists and is valid`);
}

export function errUnknownShopifyType(type: string) {
  return new Error(`Fatal Error: Unknown Shopify type '${colorType(type)}'.`);
}

export function errConfigInvalidEntry(key: string) {
  return new ConfigError(`Invalid '${colorType(key)}' configuration provided`);
}

export function errClientInvalidResult(type: ShopifyTypes) {
  return new Error(`Failed to retrieve items of type '${colorType(type)}'`);
}

export function errInvalidEntityPath(path: string) {
  return new Error(`Invalid entity path provided: '${
      colorPath(path)}' (example: 'collections/featured')`);
}

export function handlePromiseError(e: Error) {
  logging.error(e);
}