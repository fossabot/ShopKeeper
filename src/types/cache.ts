// Copyright (C) 2018 Sleep E-Z USA, Inc. / Evan Darwin
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import {GenericShopifyType} from '../http/client';
import {getOrNull} from '../lib/typing';
import {colorType, colorWarning} from '../theme';

import {ShopifyTypes} from './base';


export type StoreCacheDataElement = {
  ids: {[id: number]: number}; handles: {[handle: string]: number};
};

export type StoreCacheData = {
  indexes: {[name: string]: StoreCacheDataElement}; cache: GenericShopifyType[];
};

export const EMPTY_SHOPIFY_CACHE_DATA_ELEMENT: StoreCacheDataElement = {
  ids: {},
  handles: {}
};
export const EMPTY_SHOPIFY_CACHE: StoreCacheData = {
  indexes: {
    [ShopifyTypes.Shop]: EMPTY_SHOPIFY_CACHE_DATA_ELEMENT,
    [ShopifyTypes.Product]: EMPTY_SHOPIFY_CACHE_DATA_ELEMENT,
    [ShopifyTypes.Page]: EMPTY_SHOPIFY_CACHE_DATA_ELEMENT,
    [ShopifyTypes.ProductImage]: EMPTY_SHOPIFY_CACHE_DATA_ELEMENT,
    [ShopifyTypes.Option]: EMPTY_SHOPIFY_CACHE_DATA_ELEMENT,
    [ShopifyTypes.Variant]: EMPTY_SHOPIFY_CACHE_DATA_ELEMENT,
    [ShopifyTypes.Province]: EMPTY_SHOPIFY_CACHE_DATA_ELEMENT,
    [ShopifyTypes.Country]: EMPTY_SHOPIFY_CACHE_DATA_ELEMENT,
  },
  cache: [],
};

export class StoreCache {
  data: StoreCacheData = Object.assign({}, EMPTY_SHOPIFY_CACHE);

  _validateType(type: ShopifyTypes) {
    if (type in this.data.indexes) return;

    throw new Error(
        `Refusing to write cache for unknown ` +
        `type '${type}'. Please report this issue.`);
  }

  findById<T extends GenericShopifyType>(type: ShopifyTypes, id: number): T
      |undefined {
    this._validateType(type);

    if (id in this.data.indexes[type].ids) {
      const pos = this.data.indexes[type].ids[id];
      return this.data.cache[pos] as T;
    }

    return undefined;
  }

  findByHandle<T extends GenericShopifyType>(
      type: ShopifyTypes, handle: string): T|undefined {
    this._validateType(type);

    if (handle in this.data.indexes[type].handles) {
      const pos = this.data.indexes[type].handles[handle];
      return this.data.cache[pos] as T;
    }

    return undefined;
  }

  register<T extends GenericShopifyType>(elem: T): T {
    const id = elem.id();
    const type = elem.itemType;
    const handle = getOrNull<string>(elem.data['handle']);

    if (id < 0) {
      throw new Error(
          `Fatal: Refusing to cache item with negative ID. ` +
          `This means that an instance's placeholder object was not replaced` +
          ` before reaching the cache. Please submit an issue.`);
    }

    if (!id) {
      throw new Error(
          `Expected item ID to be resolved already: ` +
          `${colorWarning('id')} is ${colorType('null')}`);
    }

    this._validateType(type);

    const foundId = this.findById(type, id);
    const foundHandle = (handle) ? this.findByHandle(type, handle) : null;

    if (foundId || foundHandle) {
      return (foundId || foundHandle) as T;
    }

    const newPos = this.data.cache.length;

    this.data.cache.push(elem);
    this.data.indexes[type].ids[id] = newPos;

    if (handle) {
      this.data.indexes[type].handles[handle] = newPos;
    }

    return this.data.cache[newPos] as T;
  }
}