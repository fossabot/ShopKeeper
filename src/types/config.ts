// Copyright (C) 2018 Sleep E-Z USA, Inc. / Evan Darwin
// 
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
// 
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
// 
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import {PathLike} from 'fs';
import {ParsedArgs} from 'minimist';

import {RenderEngine} from '../renderers/lib';

import {Store, StoreData$Raw} from './store';

export type ConsoleCommand = (config: Config|null, args: ParsedArgs) =>
    Promise<number>;

export type Config$Templates = {
  path: PathLike,
  engines: Array<RenderEngine<{}>>
};

export type Config$Unsafe = {
  [key: string]: undefined|null|number|string|{}
};

export type Config$Stores = {
  [name: string]: Store
};
export type Config = {
  data: PathLike,
  stores: Config$Stores,
  templates: Config$Templates,
};

export type Config$User = {
  data: PathLike,
  stores: {[name: string]: StoreData$Raw},
  templates: Config$Templates,
};

export type UserConfigData = {
  pages?: null[],
};