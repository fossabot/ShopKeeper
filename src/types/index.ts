export * from './base';
export * from './cache';
export * from './config';
export * from './inventory';
export * from './routes';
export * from './store';
export * from './shopify/index';
