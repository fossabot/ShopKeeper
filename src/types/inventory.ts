// Copyright (C) 2018 Sleep E-Z USA, Inc. / Evan Darwin
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
//
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

/**
 * This value describes the inventory management service
 * set as the store default, or set on a per-product basis.
 * The default manager is Shopify itself, but you can build
 * a custom InventoryService using the Shopify API.
 *
 * Shopify's thoroughly detailed documentation is available here:
 *  https://help.shopify.com/api/reference/fulfillmentservice
 *
 * @default `"shopify"`
 */
export type InventoryManager = 'shopify'|string;

/**
 * This value describes the fulfillment service that Shopify
 * uses for fulfilling orders. By default, purchase notifications
 * will stay within the Shopify platform (`manual`), but
 * alternative fulfillment methods can be set up.
 *
 * Shopify documentation on Fulfillment Methods:
 *  https://help.shopify.com/api/reference/fulfillmentservice
 *
 * @default `"manual"`
 */
export type InventoryFulfillment = 'manual'|string;

/**
 * This property describes the behavior of how Shopify
 * will handle purchases after a `Product`'s inventory
 * has run out of stock.
 *
 * **`continue`** - Allow the user to continue checking
 * out and let the [[InventoryData.quantity]] underflow into the negatives.
 *
 * **`deny`** - Refuse to let customers purchase this item
 * until the stock is replenished.
 *
 * @default `"continue"`
 */
export type InventoryPolicy = 'continue'|'deny';

/**
 * This option describes the unit that `Product` weights
 * are defined in.
 *
 * @default `"lb"`
 */
export type ShippingWeightUnit = 'k'|'kg'|'oz'|'lb';

/**
 * This is the user-facing representation of Shopify's
 * inventory configuration options. This type should be
 * defined in user created objects where inventory options
 * are applicable ([[ProductItem]], [[VariantItem]]).
 */
export type InventoryData = {
  fulfillment: InventoryFulfillment,
  manager: InventoryManager,
  policy: InventoryPolicy,
  quantity?: number,
};

export type Shippable$Raw = {
  grams: number|null,
  weight: number|null,
  weight_unit: ShippingWeightUnit,
  requires_shipping: boolean,
  inventory_management: InventoryManager,
  fulfillment_service: InventoryFulfillment,
  inventory_policy: InventoryPolicy,
  inventory_quantity: number,
};

export type Shippable = {
  grams: number,
  weight: number,
  weightUnit: ShippingWeightUnit,
  requiresShipping: boolean,
  inventory: InventoryData,
};

export const DEFAULT_INVENTORY: InventoryData = {
  fulfillment: 'manual',
  manager: 'shopify',
  policy: 'deny',
};