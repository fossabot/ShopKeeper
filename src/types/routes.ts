// Copyright (C) 2018 Sleep E-Z USA, Inc. / Evan Darwin
// 
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU Affero General Public License as
// published by the Free Software Foundation, either version 3 of the
// License, or (at your option) any later version.
// 
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Affero General Public License for more details.
// 
// You should have received a copy of the GNU Affero General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.

import {ShopifyTypes} from './base';

export enum Method {
  List = 'list',
  Count = 'count',
  Fetch = 'fetch',
  Create = 'create',
  Update = 'update',
  Delete = 'delete',
}

export type ItemRouteMethods = {
  list: string|null,
  count: string|null,
  fetch: string|null,
  create: string|null,
  update: string|null,
  delete: string|null,
  [method: string]: string|null,
};

export const STANDARD_ITEM_ROUTE_METHODS: ItemRouteMethods = {
  list: '{prefix}.json',
  create: '{prefix}.json',
  count: '{prefix}/count.json',
  fetch: '{prefix}/{id}.json',
  update: '{prefix}/{id}.json',
  delete: '{prefix}/{id}.json',
};

export const STANDARD_ITEM_ROUTES: ItemRoutes = {
  paginated: false,
  publicPath: null,
  methods: STANDARD_ITEM_ROUTE_METHODS,
  prefix: '',
};

export type ItemRoutes = {
  /**
   * The prefix is the type as documented by Shopify's Admin
   * API. For example the PriceRule type's prefix would become
   * 'price_rule'. This is used for automatic URL resolution.
   */
  prefix: string,

  // This is a string with supported variables that will
  // automatically be replaced by ShopKeeper's engine.
  publicPath: string|null,

  // Describes which methods are available and provides
  // additional context to support Shopify's complex routing
  methods: ItemRouteMethods,

  // Describes if retrieval of all objects requires
  // pagination via the REST API.
  paginated: boolean,
};

export const GLOBAL_SHOPKEEPER_ROUTES: {[name: string]: ItemRoutes} = {
  [ShopifyTypes.Country]: {
    ...STANDARD_ITEM_ROUTES,
    prefix: 'countries',
  },

  [ShopifyTypes.Province]: {
    ...STANDARD_ITEM_ROUTES,
    prefix: 'provinces',
    methods: {
      update: 'countries/{country_id}/provinces/{id}.json',
      fetch: 'countries/{country_id}/provinces/{id}.json',
      list: 'countries/{country_id}/provinces.json',
      count: 'countries/{country_id}/provinces/count.json',
      create: null,
      delete: null,
    }
  },

  [ShopifyTypes.Product]: {
    ...STANDARD_ITEM_ROUTES,
    publicPath: '{prefix}/{handle}',
    prefix: 'products',
    paginated: true,
  },

  [ShopifyTypes.Variant]: {
    ...STANDARD_ITEM_ROUTES,
    publicPath: 'products/{handle}?variant={id}',
    prefix: 'variants',
    paginated: false,
    methods: {
      ...STANDARD_ITEM_ROUTE_METHODS,
      list: 'products/{pid}/{prefix}.json',
      count: 'products/{pid}/{prefix}/count.json',
      delete: 'products/{pid}/{prefix}/delete.json',
    },
  },

  [ShopifyTypes.Redirect]: {
    ...STANDARD_ITEM_ROUTES,
    prefix: 'redirects',
    publicPath: '{origin}',
    paginated: true,
  },

  [ShopifyTypes.Shop]: {
    ...STANDARD_ITEM_ROUTES,
    prefix: 'shop',
    methods: {
      ...STANDARD_ITEM_ROUTE_METHODS,
      fetch: '{prefix}.json',
      count: null,
      create: null,
      update: null,
      delete: null,
      list: null,
    },
  },

  [ShopifyTypes.Page]: {
    ...STANDARD_ITEM_ROUTES,
    prefix: 'pages',
    methods: STANDARD_ITEM_ROUTE_METHODS,
  },
};