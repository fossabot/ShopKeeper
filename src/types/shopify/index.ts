export * from './country';
export * from './image';
export * from './metafield';
export * from './option';
export * from './page';
export * from './product';
export * from './province';
export * from './redirect';
export * from './shop';
export * from './variant';