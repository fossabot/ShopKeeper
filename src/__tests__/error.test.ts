import {errClientInvalidResult, errConfigInvalidEntry, errInvalidEntityPath, errNoValidConfig, errUnknownShopifyType, handlePromiseError} from '../error';
import {UndefinedStackError} from '../testing/data';
import {ShopifyTypes} from '../types/base';


describe('Shared Error Functions', () => {
  it('UnknownShopifyType', () => {
    expect(() => {
      throw errUnknownShopifyType('segap');  // pages
    }).toThrowErrorMatchingSnapshot();
  });

  it('InvalidEntityPath', () => {
    expect(() => {
      throw errInvalidEntityPath('some/path');  // pages
    }).toThrowErrorMatchingSnapshot();
  });

  it('NoValidConfig', () => {
    expect(() => {
      throw errNoValidConfig();
    }).toThrowErrorMatchingSnapshot();
  });

  it('ClientInvalidResult', () => {
    expect(() => {
      throw errClientInvalidResult(ShopifyTypes.Product);
    }).toThrowErrorMatchingSnapshot();
  });

  it('ConfigInvalidEntry', () => {
    expect(() => {
      throw errConfigInvalidEntry('somekey');
    }).toThrowErrorMatchingSnapshot();
  });

  it('Default Promise Handler', () => {
    global.console.error = jest.fn();
    global.console.log = jest.fn();

    const testErr = new UndefinedStackError('Testing');

    handlePromiseError(testErr);
    expect(global.console.log).toMatchSnapshot();
    expect(global.console.error).toMatchSnapshot();
  });
});
