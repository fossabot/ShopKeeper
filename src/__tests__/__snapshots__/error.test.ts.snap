// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shared Error Functions ClientInvalidResult 1`] = `"Failed to retrieve items of type 'products'"`;

exports[`Shared Error Functions ConfigInvalidEntry 1`] = `"Invalid 'somekey' configuration provided"`;

exports[`Shared Error Functions Default Promise Handler 1`] = `[MockFunction]`;

exports[`Shared Error Functions Default Promise Handler 2`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "
[E] Testing
",
    ],
  ],
}
`;

exports[`Shared Error Functions InvalidEntityPath 1`] = `"Invalid entity path provided: 'some/path' (example: 'collections/featured')"`;

exports[`Shared Error Functions NoValidConfig 1`] = `"This command requires a valid configuration. Verify 'shopify.config.js' exists and is valid"`;

exports[`Shared Error Functions UnknownShopifyType 1`] = `"Fatal Error: Unknown Shopify type 'segap'."`;
